numpy/lib/polynomial.py

```diff
@@ -651,6 +651,8 @@ def polyval(p, x):
     for polynomials of high degree the values may be inaccurate due to
     rounding errors. Use carefully.
 
+    If `x` is a subtype of `ndarray` the return value will be of the same type.
```

Comment:
Not strictly true: `np.polyval(np.ma.array([]), 1)` does not return a masked array.

Reply:
I'm not sure what point I was trying to make here. I think what you have right now is fine.